ansible_base/authentication/apps.py

```diff
@@ -0,0 +1,8 @@
+from django.apps import AppConfig
+
+import ansible_base.checks  # noqa: F401 - register checks
+
+
+class AuthenticationConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
+    name = 'ansible_base.authentication'
```

Comment:
:+1: First thing to get out of the way is app names, and this looks great to me.

Reply:
I thought so but then I found a weird thing just now. If I run:
```
python manage.py makemigration ansible_base.authentication
```

I get this:
```
$ python manage.py makemigrations ansible_base.authentication
No installed app with label 'ansible_base.authentication'. Did you mean 'authentication'?
```

So I need to figure out where/why it thinks the app is just `authentication` and not `ansible_base.authentication`

Reply:
Yeah, let's get this taken care of. Similar to the direction @newswangerd is going in, my advice right now would be to delete `ansible_base/apps.py` and I think that might help solve some of your problems. Yes, this means deleting "ansible_base" from INSTALLED_APPS, and that has been my mental intention with this.

Reply:
I looked into this a little and I think it might just take the final name of the app as the app name. I couldn't find a way around it. If we are worried about collisions we could do `dab_authentication` or something like that to make collisions with other libraries less likely.

Reply:
@relrod and I did a test and if there is a namespace collision (because it only uses the last part of the dot notation) things will error:
```
$ python manage.py showmigrations
Traceback (most recent call last):
  File "/Users/jowestco/GIT/github/john-westcott-iv/aap-gateway/django-ansible-base/manage.py", line 22, in <module>
    main()
  File "/Users/jowestco/GIT/github/john-westcott-iv/aap-gateway/django-ansible-base/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/jowestco/Library/Python/venvs/gateway/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/Users/jowestco/Library/Python/venvs/gateway/lib/python3.11/site-packages/django/core/management/__init__.py", line 416, in execute
    django.setup()
  File "/Users/jowestco/Library/Python/venvs/gateway/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/jowestco/Library/Python/venvs/gateway/lib/python3.11/site-packages/django/apps/registry.py", line 93, in populate
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: Application labels aren't unique, duplicates: auth
```

This makes me want to prefix the apps as a safety thing.

@newswangerd @AlanCoding @relrod thoughts on this?